http/file_server.ts

```diff
@@ -123,145 +123,159 @@ export async function serveFile(
   filePath: string,
   { etagAlgorithm, fileInfo }: ServeFileOptions = {},
 ): Promise<Response> {
-  let file: Deno.FsFile;
-  if (fileInfo === undefined) {
-    [file, fileInfo] = await Promise.all([
-      Deno.open(filePath),
-      Deno.stat(filePath),
-    ]);
-  } else {
-    file = await Deno.open(filePath);
-  }
-  const headers = setBaseHeaders();
+  try {
+    let file: Deno.FsFile;
+    if (fileInfo === undefined) {
+      [file, fileInfo] = await Promise.all([
+        Deno.open(filePath),
+        Deno.stat(filePath),
+      ]);
+    } else {
+      file = await Deno.open(filePath);
+    }
```

Comment:
I think `Deno.errors.NotFound` can be thrown only from this part. How about surround only this part with `try {...} catch {...}` clause?

Reply:
Not a beautiful solution, but I think something like the below would work,
```ts
let file_: Deno.FsFile;
// ... try catch here
const file = file_!;
```

Reply:
BTW, a big `try {...} catch {...}` block blurs what is being caught in it. So it also hurts the readability in my opinion

Reply:
There, fixed! I think when I last tried, I didn't have the `throw` in the catch block.